Library/Homebrew/cmd/pull.rb

```diff
@@ -202,81 +217,68 @@ def pull
           retry
         end
 
-        safe_system "git", "checkout", "-B", bottle_branch, revision
-        pull_patch bottle_commit_url
-        safe_system "git", "rebase", branch
-        safe_system "git", "checkout", branch
-        safe_system "git", "merge", "--ff-only", "--no-edit", bottle_branch
-        safe_system "git", "branch", "-D", bottle_branch
+        safe_system "git", "checkout", "--quiet", "-B", bottle_branch, orig_revision
+        pull_patch bottle_commit_url, "bottle commit"
+        safe_system "git", "rebase", "--quiet", branch
+        safe_system "git", "checkout", "--quiet", branch
+        safe_system "git", "merge", "--quiet", "--ff-only", "--no-edit", bottle_branch
+        safe_system "git", "branch", "--quiet", "-D", bottle_branch
 
         # Publish bottles on Bintray
-        bintray_user = ENV["BINTRAY_USER"]
-        bintray_key = ENV["BINTRAY_KEY"]
-
-        if bintray_user && bintray_key
-          repo = Bintray.repository(tap)
-          changed_formulae.each do |f|
-            next if f.bottle_unneeded? || f.bottle_disabled?
-            ohai "Publishing on Bintray:"
-            package = Bintray.package f.name
-            version = f.pkg_version
-            curl "-w", '\n', "--silent", "--fail",
-              "-u#{bintray_user}:#{bintray_key}", "-X", "POST",
-              "-H", "Content-Type: application/json",
-              "-d", '{"publish_wait_for_secs": 0}',
-              "https://api.bintray.com/content/homebrew/#{repo}/#{package}/#{version}/publish"
-            bintray_fetch_formulae << f
-          end
-        else
-          opoo "You must set BINTRAY_USER and BINTRAY_KEY to add or update bottles on Bintray!"
+        unless ARGV.include? "--no-publish"
+          published = publish_changed_formula_bottles(tap, changed_formulae)
+          bintray_published_formulae.concat(published).uniq!
```

Comment:
Yeah, looks clumsy to me too. I'm just trying to minimize added code.

I also need to preserve the ordering of the components as they are added. Do you think it would be appropriate in terms of Homebrew style to add a `UniqueArray` or similar datatype?